train_general_model.py

```diff
@@ -59,26 +59,28 @@ def get_dataset(config: configs.BaseConfig):
 def main(logger, experiment_config: configs.BaseConfig) -> LightningModule:
     pl_logger = WandbLogger()
 
-    train_dataloader, val_dataloader, n_labels = get_dataset_(experiment_config)
+    train_dataloader, val_dataloader, n_labels = get_dataloaders(experiment_config)
 
     vit = EMGViT(
         image_size=experiment_config.window_size,
-        patch_size=experiment_config.base_model_config.patch_size,
+        patch_size=experiment_config.general_model_config.patch_size,
         num_classes=n_labels,
-        dim=experiment_config.base_model_config.dim,
-        depth=experiment_config.base_model_config.depth,
-        heads=experiment_config.base_model_config.heads,
-        mlp_dim=experiment_config.base_model_config.mlp_dim,
-        dropout=experiment_config.base_model_config.dropout,
-        emb_dropout=experiment_config.base_model_config.emb_dropout,
-        channels=experiment_config.base_model_config.channels,
+        dim=experiment_config.general_model_config.dim,
+        depth=experiment_config.general_model_config.depth,
+        heads=experiment_config.general_model_config.heads,
+        mlp_dim=experiment_config.general_model_config.mlp_dim,
+        dropout=experiment_config.general_model_config.dropout,
+        emb_dropout=experiment_config.general_model_config.emb_dropout,
+        channels=experiment_config.general_model_config.channels,
     )
 
     assert experiment_config.loss in ["MSELoss"], "Only MSELoss is supported for now"
 
-    pl_model = PLModel(vit, n_labels=n_labels, lr=experiment_config.pretrain.lr, n_frozen_layers=0, threshold=0.5, metric_prefix='pretrain/')
+    pl_model = PLModel(vit, n_labels=experiment_config.num_classes, lr=experiment_config.pretrain.lr, n_frozen_layers=0, threshold=0.5, metric_prefix='pretrain/')
+    
+    accelerator = 'cuda' if torch.cuda.is_available() else 'cpu'
```

Comment:
We want to make sure we don't run CPU when an accelerator is available since this happened on the cluster but we wouldn't notice, maybe we could raise a runtime error if we are not running a smoke test? something like having the accelerator being conditional like the above for smoke config, while fixed to GPU for normal configs

Reply:
I have it that it only runs on cpu in smoke and only gpu in normal configs. Guess we don't really need an accelerator in smoke